scripts/lib/CIME/case/case_setup.py

```diff
@@ -152,10 +152,15 @@ def _case_setup_impl(case, caseroot, clean=False, test_mode=False, reset=False):
 
             case.initialize_derived_attributes()
 
-            cost_per_node = 16 if case.get_value("MACH") == "yellowstone" else case.get_value("MAX_MPITASKS_PER_NODE")
+            cost_per_node = case.get_value("MAX_MPITASKS_PER_NODE")
             case.set_value("COST_PES", case.num_nodes * cost_per_node)
-            case.set_value("TOTALPES", case.total_tasks)
-            case.set_value("SMP_PRESENT", case.get_build_threaded())
+            threaded = case.get_build_threaded()
+            case.set_value("SMP_PRESENT", threaded)
+            if threaded:
+                smt_factor = max(1,int(case.get_value("MAX_TASKS_PER_NODE") / cost_per_node))
+                case.set_value("TOTALPES", case.total_tasks * case.thread_count / smt_factor)
+            else:
+                case.set_value("TOTALPES", case.total_tasks)
```

Comment:
Why count threads? Description shows:
```
<desc>total number of MPI tasks (setup automatically - DO NOT EDIT)</desc>
```

Reply:
Then I'll update the description - this is intended to be the total number of physical cores used.

Reply:
Isn't MAX_MPITASKS_PER_NODE usually the total number of available physical cores. If the cores are capable of hyper-threading (up to 4 on KNLs), then the cost may not be accurate (if NTHRDS is 2).

Reply:
that's what the smt_factor is all about.

Reply:
If MAX_MPITASKS_PER_NODE=64, and MAX_TASKS_PER_NODE=256, and the job runs on 1 node (64 cores) with 64 mpi tasks and 2 threads per task, then TOTALPES 64*2/4=32 ?

Reply:
good point - correction coming